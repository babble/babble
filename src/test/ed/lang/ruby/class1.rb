
class Foo

  def initialize
    @counter = 1;
  end
  
  def inc
    @counter += 1;
  end
  
  def jump( num )
    @counter += num;
  end
  
  def print
    puts "num clicks #{@counter}\n";
  end
end


f = Foo.new();
f.print();
f.inc();
f.print();
f.jump( 5 );
f.print();

