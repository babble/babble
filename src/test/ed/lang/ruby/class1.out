num clicks 1
num clicks 2
num clicks 7
